mllib/src/main/scala/org/apache/spark/mllib/clustering/LDA.scala

```diff
@@ -315,6 +315,27 @@ class LDA private (
     this
   }
 
+  // Initial LDAModel can be provided rather than using random initialization
+  private var initialModel: Option[LDAModel] = None
+
+  /**
+   * Set the initial starting point, bypassing the random initialization.
+   * This can be used for incremental learning.
+   * This is supported only for online optimizer, and the condition model.k == this.k must be met,
+   * failure results in an IllegalArgumentException.
+   */
+  @Since("2.2.0")
+  def setInitialModel(model: LDAModel): this.type = {
+    require(model.k == k, "mismatched number of topics")
+    this.ldaOptimizer match {
+      case _: OnlineLDAOptimizer =>
+        initialModel = Some(model)
+        this
+      case _ => throw new IllegalArgumentException(
+        "Only online optimizer supports initialization with a previous model.")
+    }
+  }
```

Comment:
I'm thinking we should move all the parameter check into run, since user may set parameters in different orders.